lib/tasks/avalon_migrations.rake

```diff
@@ -23,5 +23,33 @@ namespace :avalon do
         collection.save!(validate: false)
       end
     end
+    desc "Migrate legacy IndexedFile captions to ActiveStorage as part of supporting upload of multiple captions files"
+    task caption_files: :environment do
+      count = 0
+      # Iterate through all caption IndexedFiles
+      IndexedFile.where("id: */captions").each do |caption_file|
+        # Retrieve parent master file
+        master_file_id = caption_file.id.split('/').first
+        master_file = MasterFile.find(master_file_id) rescue nil
+        next unless master_file
```

Comment:
Maybe we also need the check that the content isn't empty?  I'm wondering if captions files that have already been migrated by this rake task would cause problems or create a bunch of empty supplemental files.  Have you run this locally after this change and seen what happens?

Reply:
Yep, good call. I had run the task once after these changes and everything looked fine but just ran the task again and it did create an empty SupplementalFile. Will add a check that there is actually a file to copy.

Reply:
Is there a way we can destroy the captions IndexedFile object?  That would be nicer but that might be overthinking it and this is probably good enough.

Reply:
I will see. I copied how the master_file controller was setup in its `delete_captions` method, but I can do some testing for a better method. Probably best practice, if possible, to have both the check for a populated IndexedFile and to fully delete them at the end.